ws/api/endpoint_test.go

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Mainflux
+// SPDX-Licence-Identifier: Apache-2.0
+
+package api_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/gorilla/websocket"
+	"github.com/mainflux/mainflux"
+
+	log "github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/ws"
+
+	httpmock "github.com/mainflux/mainflux/http/mocks"
+	"github.com/mainflux/mainflux/ws/api"
+	"github.com/mainflux/mainflux/ws/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	chanID   = "1"
+	id       = "1"
+	thingKey = "thing_key"
+	protocol = "ws"
+)
+
+var msg = []byte(`[{"n":"current","t":-1,"v":1.6}]`)
+
+func newService(cc mainflux.ThingsServiceClient) ws.Service {
+	pubsub := mocks.NewPubSub()
+	return ws.New(cc, pubsub)
+}
+
+func newHTTPServer(svc ws.Service) *httptest.Server {
+	logger := log.NewMock()
+	mux := api.MakeHandler(svc, logger)
+	return httptest.NewServer(mux)
+}
+
+func NewThingsClient() mainflux.ThingsServiceClient {
+	return httpmock.NewThingsClient(map[string]string{thingKey: chanID})
```

Comment:
No need for this wrapper, call `httpmock.NewThingsClient(map[string]string{thingKey: chanID})` directly.